bundles/org.openhab.binding.ondilo/src/main/java/org/openhab/binding/ondilo/internal/OndiloBridge.java

```diff
@@ -214,6 +209,20 @@ private void adaptPollingToValueTime(Instant lastValueTime, int refreshInterval)
                 }
                 ondiloHandler.updateRecommendationChannels(recommendation);
             }
+
+            type = new TypeToken<PoolInfo>() {
+            }.getType();
+            PoolInfo poolInfo = apiClient.request("GET", "/pools/" + id + "/device", type);
```

Comment:
I meant one small step further, can be applied to all method calls:
```suggestion
            PoolInfo poolInfo = apiClient.request("GET", "/pools/" + id + "/device", PoolInfo.class);
```

Reply:
https://stackoverflow.com/questions/9598707/gson-throwing-expected-begin-object-but-was-begin-array pointed me to the current implementation which is preferred over the `*.class`
With the `*.class` I have to use `*[].class` for lists - in this case I just receive an array, not an list which would require further changes in the code. Is it worth the change?